Plugin~/Src/MeshSyncTest/TestMeshSyncClient.cpp

```diff
@@ -18,8 +19,101 @@
 #include "MeshSync/Utility/msMaterialExt.h"     //standardMaterial
 #include "MeshSync/SceneCache/msSceneCacheInputFile.h"
 
+#include "MeshSync/AsyncSceneSender.h" //ms::AsyncSceneSender
+#include "MeshSync/msServer.h"
+
 using namespace mu;
 
+TestCase(Test_SendProperties) {
+    std::shared_ptr<ms::Scene> scene = ms::Scene::create();
+    {
+        std::shared_ptr<ms::Mesh> node = ms::Mesh::create();
+        scene->entities.push_back(node);
+
+        node->path = "/Test/PropertiesHolder";
+        node->position = { 0.0f, 0.0f, 0.0f };
+        node->rotation = quatf::identity();
+        node->scale = { 1.0f, 1.0f, 1.0f };
+        node->visibility = { false, true, true };
+        MeshGenerator::GenerateIcoSphereMesh(node->counts, node->indices, node->points, node->m_uv[0], 0.1f, 1);
+        node->refine_settings.flags.Set(ms::MESH_REFINE_FLAG_GEN_NORMALS, true);
+        node->refine_settings.flags.Set(ms::MESH_REFINE_FLAG_GEN_TANGENTS, true);
+
+        scene->propertyInfos.clear();
+        {
+            auto prop = ms::PropertyInfo::create();
+            prop->path = node->path;
+            prop->name = "Test int";
+            prop->set(10, 0, 100);
+            assert(prop->type == ms::PropertyInfo::Type::Int && "type is not correct");
+            scene->propertyInfos.push_back(prop);
+        }
+        {
+            auto prop = ms::PropertyInfo::create();
+            prop->path = node->path;
+            prop->name = "Test float";
+            prop->set(10.0f, 0, 100);
+            assert(prop->type == ms::PropertyInfo::Type::Float && "type is not correct");
+            scene->propertyInfos.push_back(prop);
+        }
+        {
+            auto prop = ms::PropertyInfo::create();
+            prop->path = node->path;
+            prop->name = "Test string";
+            const char* testString = "Test123";
+            prop->set(testString, strlen(testString));
+            assert(prop->type == ms::PropertyInfo::Type::String && "type is not correct");
+            scene->propertyInfos.push_back(prop);
+        }
+    }
+
+    TestUtility::Send(scene);
+}
+
+TestCase(Test_ServerInitiatedMessage) {
+    ms::AsyncSceneSender sender;
+    sender.requestServerInitiatedMessage();
```

Comment:
It sometimes doesn't wait and logs these exceptions instead:
```
Test_ServerInitiatedMessage begin
Exception thrown at 0x00007FFB37214FD9 in MeshSyncTest.exe: Microsoft C++ exception: Poco::Net::InvalidSocketException at memory location 0x000000363B6FDAE0.
Exception thrown at 0x00007FFB37214FD9 in MeshSyncTest.exe: Microsoft C++ exception: [rethrow] at memory location 0x0000000000000000.
Exception thrown at 0x00007FFB37214FD9 in MeshSyncTest.exe: Microsoft C++ exception: Poco::Net::InvalidSocketException at memory location 0x000000363B6FDAE0.
Exception thrown at 0x00007FFB37214FD9 in MeshSyncTest.exe: Microsoft C++ exception: Poco::Net::NetException at memory location 0x000000363B6FDFF0.
Exception thrown at 0x00007FFB37214FD9 in MeshSyncTest.exe: Microsoft C++ exception: Poco::Net::InvalidSocketException at memory location 0x000000363B6FBB60.
Test_ServerInitiatedMessage end (265.62ms)
```

I can still see the object being sent to Unity, so I have made sure that the server is active.

Reply:
It works withouut issues for me in debug mode, the destructor of the AsyncSceneSender aborts the network request, that gives me errors in the console because a thread is terminated by force.
The server is not guaranteed to be running and if Unity is running, it is not in test mode.
We'd need to set this up so it connects to a mock server in these test to properly test the client-server interaction.
Do we already have a BUILD_TESTS define? It doesn't seem to work for me, I don't see any custom defines in the test project at all.

Reply:
> Do we already have a BUILD_TESTS define?

You can setup a test project by making the project using 

```
$ cmake -G "Visual Studio 15 2017" -A x64 .. -DBUILD_TESTS=ON
```

This is written in the [BuildPlugins](https://github.com/unity3d-jp/MeshSync/blob/dev/Plugin~/Docs/en/BuildPlugins.md#tips) page.

So to run the test:
1. Create the test solution
2. Build the test solution
3. Launch MeshSync~ test project and create MeshSyncServer gameobject
4. Run the MeshSyncTest project in the solution.

Reply:
I'm building it that way already. BUILD_TESTS is not defined.
I've increased the timeout, can you test if that fixes the tests getting stuck for you?
We should not need a different timeout for tests because the request is aborted.

Reply:
It gives me these exception errors now:
```
Test_ServerInitiatedMessage begin
Exception thrown at 0x00007FFB37214FD9 in MeshSyncTest.exe: Microsoft C++ exception: Poco::IOException at memory location 0x0000008369AFD430.
Exception thrown at 0x00007FFB37214FD9 in MeshSyncTest.exe: Microsoft C++ exception: [rethrow] at memory location 0x0000000000000000.
Exception thrown at 0x00007FFB37214FD9 in MeshSyncTest.exe: Microsoft C++ exception: Poco::IOException at memory location 0x0000008369AFD430.
Exception thrown at 0x00007FFB37214FD9 in MeshSyncTest.exe: Microsoft C++ exception: Poco::Net::NetException at memory location 0x0000008369AFE250.
Exception thrown at 0x00007FFB37214FD9 in MeshSyncTest.exe: Microsoft C++ exception: Poco::IOException at memory location 0x0000008369AFBDE0.
Test_ServerInitiatedMessage end (2015.90ms)
```

Do you have these exception errors on your side ?